telemetry-core/src/main/java/com/newrelic/telemetry/transport/BatchDataSender.java

```diff
@@ -71,24 +71,25 @@ private String buildUserAgent(String additionalUserAgent) {
     return BASE_USER_AGENT_VALUE + " " + additionalUserAgent;
   }
 
-  public Response send(String json, UUID batchId)
+  public Response send(String json, UUID batchId, String batchType)
       throws DiscardBatchException, RetryWithSplitException, RetryWithBackoffException,
           RetryWithRequestedWaitException {
     if (auditLoggingEnabled) {
-      logger.debug("Sending json: " + json);
+      logger.debug("Sending json for {} : {} ", batchType, json);
     }
-    byte[] payload = generatePayload(json);
+    byte[] payload = generatePayload(json, batchType);
 
-    return sendPayload(payload, batchId);
+    return sendPayload(payload, batchId, batchType);
   }
 
-  private byte[] generatePayload(String json) throws DiscardBatchException {
+  private byte[] generatePayload(String json, String batchType) throws DiscardBatchException {
     byte[] payload;
     try {
       payload = compressJson(json);
     } catch (IOException e) {
       logger.error(
-          "Failed to serialize the batch for sending to the ingest API. Discard batch recommended.",
+          "Failed to serialize the {} for sending to the ingest API. Discard batch recommended.",
+          batchType,
           e);
```

Comment:
If you the string template placeholder (i.e. `{}`), all the arguments are interpreted parameters for the string template. So in this case, the exception will no longer be included.

Reply:
I tested this again and the exception and stack trace is logged. By "included", I believe you are referring to the exception and its stack trace being logged?

Reply:
Yes. The previous method signature being invoked was:
```
void error(String msg, Throwable t);
```
Which takes the throwable and logs information about it (i.e. the stack trace).

The new call invokes:
```
error(String format, Object arg1, Object arg2)
```
Which is different than the previous behavior. It will try to format the `format` arg using the given args, normally replacing each `{}` with `argN.toString()`.

Reply:
Ok, resolved.